sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1010,20 +1011,27 @@ public PDone expand(PCollection<T> input) {
                           getMaxBatchBytesSize(), MAX_PUBLISH_BATCH_BYTE_SIZE_DEFAULT))));
           return PDone.in(input.getPipeline());
         case UNBOUNDED:
-          return input
-              .apply(MapElements.into(new TypeDescriptor<PubsubMessage>() {}).via(getFormatFn()))
-              .apply(
-                  new PubsubUnboundedSink(
-                      getPubsubClientFactory(),
-                      NestedValueProvider.of(getTopicProvider(), new TopicPathTranslator()),
-                      getTimestampAttribute(),
-                      getIdAttribute(),
-                      100 /* numShards */,
-                      MoreObjects.firstNonNull(
-                          getMaxBatchSize(), PubsubUnboundedSink.DEFAULT_PUBLISH_BATCH_SIZE),
-                      MoreObjects.firstNonNull(
-                          getMaxBatchBytesSize(),
-                          PubsubUnboundedSink.DEFAULT_PUBLISH_BATCH_BYTES)));
+          PCollection<PubsubMessage> output =
+              input.apply(
+                  MapElements.into(new TypeDescriptor<PubsubMessage>() {}).via(getFormatFn()));
+          if (ExperimentalOptions.hasExperiment(input.getPipeline().getOptions(), "beam_fn_api")
```

Comment:
I think this should be "beam_fn_api" || "use_runner_v2" || "use_unified_worker"

Also, can we use the full pubsub with message attributes coder, this will allow for future changes to the PubsubUnboundedSink without needing to break update compatibility